contracts/ETHRegistrarController.sol

```diff
@@ -91,6 +110,37 @@ contract ETHRegistrarController is Ownable {
         }
     }
 
+    function registerWithConfig(string calldata name, address owner, uint duration, bytes32 secret, address resolver, address addr) external payable {
+        bytes32 commitment = makeCommitmentWithConfig(name, owner, secret, resolver, addr);
+        uint cost = _invalidateCommitment(name, duration, commitment);
+
+        bytes32 label = keccak256(bytes(name));
+        uint256 tokenId = uint256(label);
+
+        // Set this contract as the (temporary) owner, giving it
+        // permission to set up the resolver.
+        uint expires = base.register(tokenId, address(this), duration);
+
+        // The nodehash of this label
+        bytes32 nodehash = keccak256(abi.encodePacked(base.baseNode(), label));
+
+        // Set the resolver
+        base.ens().setResolver(nodehash, resolver);
+
+        // Configure the resolver
+        Resolver(resolver).setAddr(nodehash, addr);
```

Comment:
Perhaps only make this call if `addr` is nonzero? That way the user can still supply a resolver address if the resolver doesn't support `setAddr`, and we save gas on a pointless set.